src/main/java/betterquesting/api2/client/gui/panels/content/PanelEntityPreview.java

```diff
@@ -93,25 +94,28 @@ public IGuiRect getTransform()
 	public void drawPanel(int mx, int my, float partialTick)
     {
         if(entity == null)
-        {
             return;
-        }
-        
-        IGuiRect bounds = this.getTransform();
-        GL11.glPushMatrix();
-        RenderUtils.startScissor(new GuiRectangle(bounds));
-		
+
+		IGuiRect bounds = this.getTransform();
+		GL11.glPushMatrix();
+		RenderUtils.startScissor(new GuiRectangle(bounds));
+
 		GL11.glColor4f(1F, 1F, 1F, 1F);
-		
+
 		int sizeX = bounds.getWidth();
 		int sizeY = bounds.getHeight();
 		float scale = Math.min((sizeY/2F)/entity.height, (sizeX/2F)/entity.width);
-		float pitch = EntityList.getEntityString(entity).contains("Wisp") ? 90F : pitchDriver.readValue();
+		float pitch;
+		if (EntityList.getEntityString(entity).contains("Wisp")) {
+			Minecraft.getMinecraft().thePlayer.rotationPitch = 90F;
```

Comment:
I think you should reset the pitch immediately after rendering the wisp instead of on gui closing, to avoid a sudden pitch change visible in the background

Reply:
Well, the Wisp's model adapts to the player's point of view, so if it changes during its preview, it will adapt to its changes.

As far as I understand, the Wisp does not have a real 3D model. It has a 2D effect that is orthogonal to the player's point of view, in which case it is always completely visible to the player.
That is why I change the player's pitch when the GUI opens, and it stays like that until preview is not needed anymore, so it is safe to change it back, which is when you close the Wisp's preview.

An alternative might be extending the relevant class and overriding method that follows the player's pitch.
Do we have Thaumcraft's source code?

Reply:
I mean that you can set the pitch to 90, draw the wisp and then reset the pitch to the saved value so the player's visible perspective never changes, you can see it outside of the BQ window border

Reply:
I suspect that this from `thaumcraft.client.renderers.entity.RenderWisp`:
```
public void renderEntityAt(Entity entity, double x, double y, double z, float fq, float pticks) {
  if(((EntityLiving)entity).getHealth() > 0.0F) {
     float f1 = ActiveRenderInfo.rotationX;
     float f2 = ActiveRenderInfo.rotationXZ;
     float f3 = ActiveRenderInfo.rotationZ;
     float f4 = ActiveRenderInfo.rotationYZ;
     float f5 = ActiveRenderInfo.rotationXY;
```
which uses `net.minecraft.client.renderer.ActiveRenderInfo` where in method:
```
 /** Updates the current render info and camera location based on entity look angles  and 1st/3rd person view mode*/
public static void updateRenderInfo(EntityPlayer p_74583_0_, boolean p_74583_1_) {
    ...
    float f2 = p_74583_0_.rotationPitch;
    float f3 = p_74583_0_.rotationYaw;
    rotationX = MathHelper.cos(f3 * (float)Math.PI / 180.0F) * (float)(1 - i * 2);
    rotationZ = MathHelper.sin(f3 * (float)Math.PI / 180.0F) * (float)(1 - i * 2);
    rotationYZ = -rotationZ * MathHelper.sin(f2 * (float)Math.PI / 180.0F) * (float)(1 - i * 2);
    rotationXY = rotationX * MathHelper.sin(f2 * (float)Math.PI / 180.0F) * (float)(1 - i * 2);
    rotationXZ = MathHelper.cos(f2 * (float)Math.PI / 180.0F);
}
```
might be the problem.

Yet again, it is just a guess, but at this point I have already spent too much time on this and if suggested "easy" solution is not acceptable I suggest to revert mod back to the 3.3.24 version.

Reply:
Ok, so you could do the thing I suggested, but also call `ActiveRenderInfo.updateRenderInfo` after you set the pitch to 90, and then call it again after you reset it - this should work